analysis/dataset_summary.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2023 OpenMined.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains dataset summary and the computation of the summary."""
+
+import pipeline_dp
+import dataclasses
+from typing import Iterable
+
+
+@dataclasses.dataclass
+class PublicPartitionsSummary:
+    num_dataset_public_partitions: int
+    num_dataset_non_public_partitions: int
+    num_empty_public_partitions: int
+
+
+_DATASET_PUBLIC = 1
```

Comment:
Initially I did that. But for performance reasons I decided to use plain ints (because here we need to process amount of data roughly equivalent to raw dataset). Anyway, it's internal stuff, not exposed to API.